lib/download.js

```diff
@@ -31,7 +34,72 @@ function getRequestOptions(options) {
   return options;
 }
 
+function fetchFromFilesystem(src) {
+  var fsDfd = Q.defer();
+
+  // We don't need any options here, so we just grab the url
+  // and make sure it fits the norms for our platform.
+  var givenFilePath = path.normalize(src.url);
+
+  function notify(state) {
+    fsDfd.notify(state);
+  }
+
+  var filePath;
+
+  // Test if the file path is absolute. If not, then prepend the application
+  // directory beforehand, so it can be treated as relative from the app root.
+  if (path.isAbsolute(givenFilePath)) {
+    filePath = givenFilePath;
+  } else {
+    filePath = path.join(workingDir, givenFilePath);
+  }
+
+  console.log("Retrieving `" + filePath + "`");
+
+  // Getting the errors and size right on the progress bar here is not worth all the setup.
+  // Just fail silently here if there's a problem and the error will correctly bubble in the
+  // readFile call below.
+  var totalSize = 0;
+  try {
+    var stats = fs.statSync(filePath);
+    totalSize = stats.size;
+  } catch (e) {}
+
+
+  // We're gonna go from 0->100 with nothing between for fs.readFile. We could alternatively
+  // do a readStream, but it seems like overkill for the filesizes.
+  notify({total: totalSize, received: 0, percent: 0});
+
+  // Async request the file
+  fs.readFile(filePath, function(error, fileBody) {
+      if (error) {
+        error.message = "Error retrieving file from disk.\n" + error.stack + "\n\n" +
+          "Please report this full log at " +
+          "https://github.com/rxaviers/cldr-data-downloader";
+        fsDfd.reject(error);
+        throw error;
+      } else {
+        notify({total: totalSize, received: fileBody.length, percent: 100});
+        fsDfd.resolve(fileBody);
+      }
+  });
+
+  return fsDfd.promise;
+}
+
 function download(src) {
+  // Modify the src to not be a file protocol uri so we can unify handling files
+  if (src.url && src.url.trim().toLowerCase().indexOf("file://") === 0) {
```

Comment:
I believe `src.url` is always present considering this function is only called by `/index.js` (it seems it is). Also, I believe it's fine if we stick with the .trim() version of this string.

Also, what do you think of using .replace instead of this block?

Therefore, a suggestion:

```diff
diff --git a/lib/download.js b/lib/download.js
index 7508999..15ed2de 100644
--- a/lib/download.js
+++ b/lib/download.js
@@ -89,11 +89,11 @@ function fetchFromFilesystem(src) {
 }
 
 function download(src) {
-  // Modify the src to not be a file protocol uri so we can unify handling files
-  if (src.url && src.url.trim().toLowerCase().indexOf("file://") === 0) {
-    // Chop off the first 7 characters
-    src.url = src.url.trim().substr("file://".length);
-  }
+  src.url = src.url.trim();
+
+  // Modify the url to not be a file protocol uri so we can unify handling files.
+  src.url = src.url.replace(/^file:\/\//i, "");
+
   // Short circuit the download function and grab it from the filesystem if
   // we don't have anything that looks like a protocol (e.g. https/http/ftp/etc).
   if (src.url && src.url.indexOf("://") === -1) {
```